pkg/channel/distributed/receiver/producer/producer_test.go

```diff
@@ -18,124 +18,64 @@ package producer
-
-	TestConfigMetadataChange = TestConfigAdmin + TestConfigNet + `
-Metadata:
-  RefreshFrequency: 200000` + TestConfigConsumer + TestConfigProducer
-
-	TestConfigProducerChange = TestConfigAdmin + TestConfigNet + TestConfigMeta + TestConfigProducer + `
-Producer:
-  MaxMessageBytes: 300` + TestConfigConsumer
-
-	TestConfigConsumerChange = TestConfigAdmin + TestConfigNet + TestConfigMeta + TestConfigProducer + TestConfigConsumer + `
-  Fetch:
-    Min: 200
-`
-
-	TestConfigAdminChange = `
-Admin:
-  Retry:
-    Max: 100` + TestConfigNet + TestConfigMeta + TestConfigProducer + TestConfigConsumer
-
-	TestSaramaConfigYaml = TestConfigAdmin + `
-Net:
-  TLS:
-    Config:
-      ClientAuth: 0
-  SASL:
-    Mechanism: PLAIN
-    Version: 1
-    User: ` + receivertesting.KafkaUsername + `
-    Password: ` + receivertesting.KafkaPassword + TestConfigMeta + TestConfigConsumer + `
-ClientID: ` + receivertesting.ClientId + `
-`
-
-	TestEventingKafka = `
-kafka:
-  enableSaramaLogging: true`
 )
 
 // Test The NewProducer Constructor
 func TestNewProducer(t *testing.T) {
 
+	// Test Data
+	brokers := []string{receivertesting.KafkaBroker}
+	config := sarama.NewConfig()
+
 	// Create A Mock Kafka SyncProducer
-	mockSyncProducer := receivertesting.NewMockSyncProducer()
+	mockSyncProducer := producertesting.NewMockSyncProducer()
 
-	// Create A Test Producer
-	producer := createTestProducer(t, mockSyncProducer)
+	// Stub NewSyncProducerWrapper() For Testing And Restore After Test
+	producertesting.StubNewSyncProducerFn(producertesting.ValidatingNewSyncProducerFn(t, brokers, config, mockSyncProducer))
+	defer producertesting.RestoreNewSyncProducerFn()
 
-	// Verify The Results
-	assert.True(t, producer.healthServer.ProducerReady())
+	// Create And Validate A Test Producer
+	createTestProducer(t, brokers, config, mockSyncProducer)
+
+	// Verify The Mock SyncProducer State
+	assert.False(t, mockSyncProducer.Closed())
 }
 
 // Test The ProduceKafkaMessage() Functionality For Event With PartitionKey
 func TestProduceKafkaMessage(t *testing.T) {
 
-	// Create Test Data
-	mockSyncProducer := receivertesting.NewMockSyncProducer()
-	producer := createTestProducer(t, mockSyncProducer)
+	// Test Data
+	brokers := []string{receivertesting.KafkaBroker}
+	config := sarama.NewConfig()
 	channelReference := receivertesting.CreateChannelReference(receivertesting.ChannelName, receivertesting.ChannelNamespace)
 	bindingMessage := receivertesting.CreateBindingMessage(cloudevents.VersionV1)
 
+	// Create A Mock Kafka SyncProducer
+	mockSyncProducer := producertesting.NewMockSyncProducer()
+
+	// Stub NewSyncProducerWrapper() For Testing And Restore After Test
+	producertesting.StubNewSyncProducerFn(producertesting.ValidatingNewSyncProducerFn(t, brokers, config, mockSyncProducer))
+	defer producertesting.RestoreNewSyncProducerFn()
+
+	// Create Producer To Test
+	producer := createTestProducer(t, brokers, config, mockSyncProducer)
```

Comment:
Yeah, I've had them combined before (removed the test) and it can sometimes get messy when you want to verify different things... You are right about similarity but I'd like to leave it this way for now.